components/os/component.go

```diff
@@ -3,140 +3,365 @@ package os
+	ctx         context.Context
+	cancel      context.CancelFunc
+	eventStore  eventstore.Store
+	eventBucket eventstore.Bucket
+
+	lastMu   sync.RWMutex
+	lastData *Data
+}
 
-func New(ctx context.Context, cfg Config, eventStore eventstore.Store) (components.Component, error) {
-	eventBucket, err := eventStore.Bucket(os_id.Name)
+func New(ctx context.Context, eventStore eventstore.Store) (components.Component, error) {
+	eventBucket, err := eventStore.Bucket(Name)
 	if err != nil {
 		return nil, err
 	}
 
-	cfg.Query.SetDefaultsIfNotSet()
-	setDefaultPoller(cfg, eventBucket)
-
 	cctx, ccancel := context.WithCancel(ctx)
-	getDefaultPoller().Start(cctx, cfg.Query, os_id.Name)
-
 	return &component{
-		rootCtx:      ctx,
-		cancel:       ccancel,
-		poller:       getDefaultPoller(),
-		eventsBucket: eventBucket,
-		cfg:          cfg,
+		ctx:         cctx,
+		cancel:      ccancel,
+		eventStore:  eventStore,
+		eventBucket: eventBucket,
 	}, nil
 }
 
-var _ components.Component = &component{}
+func (c *component) Name() string { return Name }
 
-type component struct {
-	rootCtx      context.Context
-	cancel       context.CancelFunc
-	poller       query.Poller
-	eventsBucket eventstore.Bucket
-	cfg          Config
+func (c *component) Start() error {
+	go func() {
+		ticker := time.NewTicker(time.Minute)
+		defer ticker.Stop()
+		for {
+			c.CheckOnce()
+			select {
+			case <-c.ctx.Done():
+				return
+			case <-ticker.C:
+			}
+		}
+	}()
+	return nil
+}
+
+func (c *component) States(ctx context.Context) ([]components.State, error) {
+	c.lastMu.RLock()
+	lastData := c.lastData
+	c.lastMu.RUnlock()
+	return lastData.getStates()
 }
 
-func (c *component) Name() string { return os_id.Name }
+func (c *component) Events(ctx context.Context, since time.Time) ([]components.Event, error) {
+	return c.eventBucket.Get(ctx, since)
+}
 
-func (c *component) Start() error { return nil }
+func (c *component) Metrics(ctx context.Context, since time.Time) ([]components.Metric, error) {
+	log.Logger.Debugw("querying metrics", "since", since)
 
-func (c *component) States(ctx context.Context) ([]components.State, error) {
-	last, err := c.poller.Last()
-	if err == query.ErrNoData { // no data
-		log.Logger.Debugw("nothing found in last state (no data collected yet)", "component", os_id.Name)
-		return []components.State{
-			{
-				Name:    os_id.Name,
-				Healthy: true,
-				Reason:  query.ErrNoData.Error(),
-			},
-		}, nil
+	return nil, nil
+}
+
+func (c *component) Close() error {
+	log.Logger.Debugw("closing component")
+
+	c.cancel()
+	c.eventBucket.Close()
+
+	return nil
+}
+
+// CheckOnce checks the current pods
+// run this periodically
+func (c *component) CheckOnce() {
+	log.Logger.Infow("checking memory")
+	d := Data{
+		ts: time.Now().UTC(),
 	}
+	defer func() {
+		c.lastMu.Lock()
+		c.lastData = &d
+		c.lastMu.Unlock()
+	}()
+
+	cctx, ccancel := context.WithTimeout(c.ctx, 15*time.Second)
+	virtEnv, err := pkg_host.SystemdDetectVirt(cctx)
```

Comment:
We move all these static information (such as virtEnv, kernel version, SystemManufacturer, reboot time, host ID, arch, etc.) into separate packages and initialize them when gpud starts.